feat(signs)!: change drawing direction of signs (#27781)
neovim/neovim: runtime/doc/news.txt

EDITOR

> • Signs are now drawn from left to right

Reviewer: need to say a bit about the motivation, otherwise it just sounds random
```suggestion
• Signs are drawn from left to right, so the "highest priority" appear leftmost, while any signs exceeding the signcolumn width (thus not displayed) are the also the lowest priority.
```

also need to mention this in vim_diff.txt

Reviewer 2: A bit misleading, they were always "drawn" left to right, what changes here is the sort order. Also which signs are visible/not visible does not change here. How about:

```suggestion
• The order in which signs are placed was changed. Higher priority signs will now appear left of lower priority signs.
```

What we already have in vim_diff.txt also does not seem like it needs to change; there is no order in Vim:

      'signcolumn'  supports up to 9 dynamic/fixed columns

Mentionioning the intention of the sort order somewhere `sign.txt` or at `:h signcolumn` in seems more relevant.

> • Signs are drawn from left to right, so the "highest priority" appear leftmost, while any signs exceeding the signcolumn width (thus not displayed) are the also the lowest priority.

Reviewer: ```suggestion
• Signs are drawn from left to right, so the "highest priority" appear leftmost, while any signs exceeding the signcolumn width (thus not displayed) are also the lowest priority.
```

Reviewer 2: A bit misleading, they were always "drawn" left to right, what changes here is the sort order. Also which signs are visible/not visible does not change. How about:

```suggestion
• The order in which signs are placed was changed. Higher priority signs will now appear left of lower priority signs.
```

What we already have in vim_diff.txt also does not seem like it needs to change; there is no order in Vim:

      'signcolumn'  supports up to 9 dynamic/fixed columns

Mentionioning the intention of the sort order somewhere `sign.txt` or at `:h signcolumn` in seems more relevant.

Reviewer 3: There were indeed drawn from right to left before, but aligned to the left.

Reviewer 2: I mean it's a technical nit but my point is that what changes is the order of the placed signs, not the order in which signs are drawn.

Reviewer 3: > Mentionioning the intention of the sort order somewhere `sign.txt` or at `:h signcolumn` in seems more relevant.

So should that go under `*sign-priority*` or maybe a new `*sign-order*` section?

Reviewer 2: `*sign-priority*`seems fine to me, that text seems unadjusted for multi-width signcolumn anyways. Maybe update to something like this:

```diff
diff --git a/runtime/doc/sign.txt b/runtime/doc/sign.txt
index 0360ce67f..048f37023 100644
--- a/runtime/doc/sign.txt
+++ b/runtime/doc/sign.txt
@@ -72,9 +72,10 @@ on the same line, the attributes of the sign with the highest priority is used
 independently of the sign group. The default priority for a sign is 10. The
 priority is assigned at the time of placing a sign.
 
-When two signs with the same priority are present, and one has an icon or text
-in the signcolumn while the other has line highlighting, then both are
-displayed.
+When multiple signs that each have an icon or text are present, signs are
+ordered with increasing priority from left to right, up until the maximum
+width set in 'signcolumn'. Lower priority signs that do not fit are hidden.
+Highest priority signs with highlight attributes are always shown.
 
 When the line on which the sign is placed is deleted, the sign is removed along
 with it.
```
